service/metric/middleware.go

```diff
@@ -0,0 +1,66 @@
+package metric
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+)
+
+// CollectAPIStats middleware collects different statistics from the HTTP request
+// use this middleware like this:
+// router := mux.NewRouter()
+// router.Handle("/v1/models", metric.CollectAPIStats("modelList", http.HandlerFunc(model.List))).Methods("GET")
+func CollectAPIStats(view string, inner http.Handler) http.Handler {
```

Comment:
This is the basic HTTP middleware to collect some stats from the HTTM request/response. It can be added to every HTTP router